src/Client/VirgilAgent.ts

```diff
@@ -0,0 +1,104 @@
+import { OS_LIST } from '../Lib/platform/osList';
+import { BROWSER_LIST } from '../Lib/platform/browserList';
+
+/**
+ * Class responsible for tracking basic information which Virgil Product is using, his version,
+ * browser and platform.
+ */
+export class VirgilAgent {
+
+	value: string;
+	private userAgent: string;
+	/**
+	 * Returns navigator.userAgent string or '' if it's not defined.
+	 * @return {string} user agent string
+	 */
+
+	private getUserAgent(): string {
+		if (typeof navigator !== "undefined" && typeof navigator.userAgent === "string") {
+			return navigator.userAgent;
+		} else {
+			return "";
+		}
+	}
+
+	/**
+	 * Initializes a new instance of `Connection`.
+	 * @param {string} product - name of product eg (sdk, brainkey, bpp, keyknox, ratchet, e3kit, purekit)
+	 * argument of request methods.
+	 * @param {string} version - version of the product.
+	 * @param {string} userAgent - string with device user agent. Optional
+	 */
+	constructor(
+		product: string,
+		version: string,
+		userAgent?: string
+	) {
+		this.userAgent = userAgent || this.getUserAgent();
+		this.value = `${product};js;${this.getHeaderValue()};${version}`;
+	};
+
+	/**
+	 * Detects device OS
+	 * @returns {string} returns OS if detected or 'other'.
+	 */
+	getOsName() {
+		const os = OS_LIST.find((os) => {
+			if (os.test.some(condition => condition.test(this.userAgent))) return true;
```

Comment:
why not just:
```suggestion
			return os.test.some(condition => condition.test(this.userAgent)))
``` 
`some` already returns boolean